File: datflix_frontend/lib/models/movie.dart
class Movie {
  final String id;
  final String title;
  final String? posterUrl;
  final List<String>? genreIds;
  final int? releaseYear;
  final String? description;
  final String? trailerUrl;
  final String? videoUrl;

  Movie({
    required this.id,
    required this.title,
    this.posterUrl,
    this.genreIds,
    this.releaseYear,
    this.description,
    this.trailerUrl,
    this.videoUrl,
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: json['id'].toString(),
      title: json['title'] ?? '',
      posterUrl: json['posterUrl'] != null
          ? json['posterUrl'].toString().startsWith('http')
          ? json['posterUrl']
          : 'http://192.168.1.102:8000${json['posterUrl']}'
          : null,
      genreIds: json['genreIds'] != null
          ? List<String>.from(json['genreIds'].map((e) => e.toString()))
          : null,
      releaseYear: json['releaseYear'] != null
          ? int.tryParse(json['releaseYear'].toString())
          : null,
      description: json['description']?.toString(),
      trailerUrl: json['trailerUrl']?.toString(),
      videoUrl: json['videoUrl']?.toString(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'posterUrl': posterUrl,
    'genreIds': genreIds,
    'releaseYear': releaseYear,
    'description': description,
    'trailerUrl': trailerUrl,
    'videoUrl': videoUrl,
  };
}

File: datflix-backend/src/main/kotlin/com/datflix/models/Movie.kt
package com.datflix.models

import kotlinx.serialization.Serializable

@Serializable
data class Movie(
    var id: String? = null,
    var title: String? = null,
    var description: String? = null,
    var releaseYear: Int? = null,
    var durationMinutes: Int? = null,
    var posterUrl: String? = null,
    var trailerUrl: String? = null,
    var videoUrl: String? = null,
    var genreIds: List<String>? = null,
    var averageRating: Float = 0f,
    var isTrending: Boolean = false
)
